ext/ext/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+// lib.rs
+
+// Copyright (C) 2020-2021 The Nitrocli Developers
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use std::env;
+use std::ffi;
+use std::process;
+
+use anyhow::Context as _;
+
+#[derive(Debug)]
+pub struct Context {
+  pub nitrocli: ffi::OsString,
+  pub resolved_usb_path: Option<String>,
+  pub verbosity: Option<u8>,
+  pub project_dirs: directories::ProjectDirs,
+}
+
+impl Context {
+  pub fn from_env(name: &str) -> anyhow::Result<Self> {
```

Comment:
Would you mind adding a method header? The `name` may not be self explanatory without looking at the implementation, I'd say.

Reply:
You mean a doc comment?  Sure, we probably should do that for the entire module.  I also wasn’t sure if we should automatically deduce the name from `env::args()` or `env::current_exe()`, or at least provide a method that does that.  What do you think?

Reply:
Yeah. Automatic deduction seems fine to me. Let's do just that for now, as it's all we need.